### PR TITLE
Add Move Files

### PR DESCRIPTION
**Pull request by ruslanpashkov**

### CODE REVIEW

File: src/app.js [hunk 1]
```diff
@@ -1,21 +1,35 @@
...
+
+    return;
+  }
+
+  const finalDestinationPath = destinationPath.endsWith('/')
+    ? path.join(destinationPath, path.basename(sourcePath))
+    : destinationPath;
+
+  fs.rename(sourcePath, finalDestinationPath);
+  console.log('File moved successfully.');
```
Comment (roman-mirzoian) [inline src/app.js:30]: If we rename the file, but do not move it, this message will not be correct.

File: src/app.js [hunk 2]
```diff
@@ -1,21 +1,35 @@
...
 }
 
-module.exports = sum;
+const [source, destination] = process.argv.slice(2);
```
Comment (roman-mirzoian) [inline src/app.js:33]: If we're just renaming the file, then the name `destination` won't quite fit, it's better to call it `param` or something similar to something more general.

File: src/app.js [hunk 3]
```diff
@@ -1,21 +1,35 @@
...
+
+moveFile(source, destination);
```
Comment (roman-mirzoian) [inline src/app.js:35]: It is not necessary to pass these variables as function arguments (`source`, `destination`), they can be defined at the beginning of the file and used directly in the function.

File: src/app.js [hunk 4]
```diff
@@ -1,21 +1,35 @@
...
+
+  const directory = path.dirname(destinationPath);
+
+  if (!fs.existsSync(directory)) {
+    console.log('Destination directory does not exist.');
+
+    return;
+  }
+
+  const finalDestinationPath = destinationPath.endsWith('/')
```
Comment (roman-mirzoian) [inline src/app.js:25]: In the new path for the file `/` will not necessarily be at the end (see the example in the task), so you need to check if it is there at all.

**Review by roman-mirzoian — CHANGES_REQUESTED** [review]
The application is well written, but now there are some errors when using it:
1. If you try to rename the file, the application will crash.
image
2. If you try to move the file to a non-existent folder, it will be renamed.
image
image
3. If you try to move the file to an existing folder, you will receive a message that everything is successful, and then an error will occur.
image

Also, the type of operation (changing the name or moving) should be done at the beginning of the function, so as not to perform unnecessary actions (for example, checking if a folder exists when we only want to change the name).

**Review by naziamoff — APPROVED** [review]
👍

**Pull request closed without merging**